### PR TITLE
[Tablet support M3] Part 2 – Toolbar refactoring

### PR DESCRIPTION
**Pull request by samiuelson**
Closes: #10615 

### Description

This PR refactors toolbar implementation in order creation screen. We are switching from MainActivity AppBarLayout to local Toolbar components to support two-pane screen mode with separate titles, menus and navigation icons.

### Testing instructions

* Create new order and observe new toolbar layout on large screens based on the designs – 6kkCwI9VOEYcidp6UT5bln-fi-192%3A25891

* On the standard screen, the toolbar should look as before, however, the collapsing app bar layout behavior is removed.

Note that the toolbar of product selector fragment will be removed in the next PR.

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4527432/0d095fec-6ee9-4876-9ea9-ee27026481fb

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/res/layout/fragment_order_create_edit_form.xml [hunk 1]
```diff
@@ -5,33 +5,41 @@
     android:layout_height="match_parent"
     xmlns:tools="http://schemas.android.com/tools">
 
+    <androidx.appcompat.widget.Toolbar
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/res/layout/fragment_order_create_edit_form.xml:8]: ❓ Can we use `MaterialToolbar` here instead?

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @samiuelson. Overall, it works fine. However I noticed couple of things worth fixing in the future PRs.

1. The back navigation behavior on the product selector screen, within a two-pane layout, does not function as anticipated when navigating within the product selector screen.


[m3_filter_toolbar.webm](https://github.com/woocommerce/woocommerce-android/assets/1331230/483a00ea-7215-4fea-bf51-a1af127d6a66)



2. In the two-pane layout of the product selector screen, there appears to be an issue with having two toolbars. These should be consolidated into a single toolbar featuring a unified navigation icon.

Screenshot 2024-02-09 at 2 39 10 PM

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/193">17.4</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>DeviceType</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>ProductSelectorSharedViewModel</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**